.travis.yml

```diff
@@ -2,6 +2,8 @@ language: python
 python:
   - "2.7"
   - "3.6"
+  - "3.7"
+dist: xenial
```

Comment:
This changes all test runs to xenial which is generally available on Travis but is still not the default distro.  A more conservative alternative would be to only run Python 3.7 on xenial:
```
matrix:
  include:
    - python: “3.7”
       dist: xenial
```

Reply:
:+1: This would indeed be a more cautious approach.

I've opted for not using the build matrix because it requires specifying all combinations of `env` in the matrix, just adding a single include will only run the first `env`. This creates a lot of duplication in the yaml file and makes it significantly bigger. In addition, trusty is end of life April 2019, which may or may not affect it's use by Travis as the default VM.

Because of the above two reasons I've opted to switch to xenial by default. It gives us py37 and keeps the current yaml file mostly intact.

That being said, I'm happy to amend this approach and run py37 on xenial and continue running py 3.6 and 2.7 on trusty if necessary.

Reply:
@purificant cant you use [global variables](https://docs.travis-ci.com/user/environment-variables/#global-variables) here?

Reply:
@cl1ent It's the same issue when using matrix inside the env, only the first `env` is run for py37 and not all permutations. My sample code was:
```
language: python
matrix:
  include:
    - python: "2.7"
    - python: "3.6"
    - python: "3.7"
      dist: xenial
# command to install dependencies
cache:
  - pip
install:
  - "pip install --upgrade pip; pip install --upgrade setuptools; pip install -r test_requirements.txt; pip install -e git+https://github.com/django/django-contrib-comments.git#egg=django-contrib-comments; python setup.py install" 
# command to run tests
env:
  matrix:
    - TESTCASE=tests/tests_docs.py
    - TESTCASE=tests/test_handler.py
    - TESTCASE=tests/tests_middleware.py
    - TESTCASE=tests/tests_placebo.py
    - TESTCASE=tests/tests_async.py
    - TESTCASE=tests/tests.py
script: 
  - nosetests $TESTCASE --with-coverage --cover-package=zappa --with-timer
    #  - coverage combine --append
after_success:
  coveralls
notifications:
  slack: zappateam:TTJ0mfHunDK0IBweKkEXjGpR
```